hermes/management/commands/hopingest.py

```diff
@@ -0,0 +1,215 @@
+        parser.add_argument('-t', '--test', required=False, default=False, action='store_true',
+                            help='Log four sys.heartbeat and exit')
+
+    def _get_hop_authentication(self, options):
+        """Get username and password and configure Hop client authentication
+
+        If command line arguments are supplied, use them.
+        Otherwise, get HOP_USERNAME and HOP_PASSWORD from environment.
+
+        For the moment, since there is not yet user account support,
+        HOP_USERNAME and HOP_PASSWORD should be set to enter the environment
+        up as k8s secrets.
+        """
+        username = options.get('username')
+        if username is None:
+            username = os.getenv('HOP_USERNAME', None)
+        password = options.get('password')
+        if password is None:
+            password = os.getenv('HOP_PASSWORD', None)
+
+        if username is None or password is None:
+            error_message = 'Supply Hop credentials on command line or set HOP_USERNAME and HOP_PASSWORD environment variables.'
+            logger.error(error_message)
+            raise CommandError(error_message)
+
+        return Auth(username, password)
+
+
+    def handle(self, *args, **options):
+        logger.debug(f'args: {args}')
+        logger.debug(f'options: {options}')
+
+        # interpret command line options
+        hop_auth = self._get_hop_authentication(options)
+        if options['test']:
+            logger.info('testing...')
+            self._test_sys_heartbeat(hop_auth)
+            exit()
+
+        start_position = StartPosition.LATEST
+        if options['earliest']:
+            start_position = StartPosition.EARLIEST
+        logger.info(f'hop.io.StartPosition set to {start_position}')
+
+        topics = options['topic']
+        if topics is None:
+            topics = [['sys.heartbeat']] # default defined here
+        logger.info(f'topic set to  {topics}')
+
+        stream_url = f'kafka://kafka.scimma.org/{",".join(topics[0])}'
+        logger.info(f'stream_url:  {stream_url}')
+
+        # map from topic to alert parser/db-updater for that topic
+        alert_handler = {
+            'gcn.circular': self._update_db_with_gcn_circular,
+            'tomtoolkit.test': self._update_db_with_hermes_alert,
+            'hermes.test': self._update_db_with_hermes_alert,
+            'sys.heartbeat': self._heartbeat_handler
+        }
+
+        # instanciate the Stream in a way that sets the io.StartPosition
+        stream = Stream(auth=hop_auth, start_at=start_position)
+        stream_url = f'kafka://kafka.scimma.org/{",".join(topics[0])}'
+        logger.info(f'stream_url:  {stream_url}')
+        with stream.open(stream_url, 'r') as src:
+            for alert, metadata in src.read(metadata=True):
+                # type(gcn_circular) is <hop.models.GNCCircular>
+                # type(metadata) is <hop.io.Metadata>
+                alert_handler[metadata.topic](alert, metadata)
```

Comment:
should we have a try/except along with a `_update_db_with_generic_alert` for scaling purposes?

Reply:
in case we pull in topics we haven't prepared for.